src/app/shared/components/button/button.component.ts

```diff
@@ -1,16 +1,28 @@
-import { Component } from '@angular/core';
+import { Component, Input } from '@angular/core';
 import { FaIconLibrary } from '@fortawesome/angular-fontawesome';
+import { IconName, IconProp } from '@fortawesome/fontawesome-svg-core';
 import { fas } from '@fortawesome/free-solid-svg-icons';
 
 @Component({
   selector: 'app-button',
   templateUrl: './button.component.html',
   styleUrls: ['./button.component.scss']
 })
-export class ButtonComponent {
+export class ButtonComponent{
+  @Input() buttonText?:string;
+  @Input() iconName?:'trash-can' | 'pencil';
+  icon: IconProp | undefined;
+  
   constructor(library: FaIconLibrary) {
     library.addIconPacks(fas);
   }
 
+  ngOnInit() {
```

Comment:
1, The Component does not implement the OnInit interface, this should be added even if it works without that.
2, However it seems more appropriate to handle the `iconName` in the `OnChanges` lifecycle hooks since it is an `Input()` of the Component. Stay tuned for lifecycle hooks....

Reply:
Changed, Thank you!